defog/query.py

```diff
@@ -196,3 +202,10 @@ def execute_query(
                 write_logs(str(e))
                 retries -= 1
         raise Exception(err_msg)
+
+
+def is_connection_error(err_msg: str) -> bool:
```

Comment:
Thank you! This works very well for Postgres, though may not work as well for Snowflake and other flavors

```
250001 (08001): Failed to connect to DB. Verify the account name is correct: [`REDACTED`/). HTTP 403: Forbidden
```

MySQL
```
1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
```

etc. Can fix later, though!